openedx/openedx-platform
# feat: added banner message for survey report

This PR adds a custom banner message to the admin site. Right now the message only appears if a survey report hasn't been generated in 6 months but this threshold is configurable . Also this PR will send the report when it's generated as well.

**How to test**

1. Ensure that your latest Survey Report generated is at least one month old (if you don't have any, then this should work as well). You can check your previous survey reports at /admin/survey_report/surveyreport/
2. If you're a super user with access to the LMS admin dashboard, you should be able to see the banner message. In any site belonging to the admin site.
![Screenshot from 2023-11-02 15-00-44](https://github.com/openedx/edx-platform/assets/79876430/d186861a-fa4f-4943-8e54-6b384cf52af6)
3. You can click on the "Send report" button to generate and send the report accordinly.
4. A "thank you message" should confirm that the report was sent properly. 
![Screenshot from 2023-11-02 15-06-10](https://github.com/openedx/edx-platform/assets/79876430/f31c0bfd-7dba-4bf3-9ff2-ace89555202e)
5. You can also click on the "Dismiss" button to hide the message.
6. The newly generated report should appear at /admin/survey_report/surveyreport/
7. The message should not appear again since the report was recently generated and sent.
